openpype/hosts/maya/plugins/publish/validate_gltf_textures_names.py

```diff
@@ -0,0 +1,136 @@
+
+    def process(self, instance):
+        """Process all the nodes in the instance"""
+        pbs_shader = cmds.ls(type="StingrayPBS")
+        if not pbs_shader:
+            raise RuntimeError("No PBS Shader in the scene")
+        invalid = self.get_texture_shader_invalid(instance)
+        if invalid:
+            raise RuntimeError("Non PBS material found"
+                               "{0}".format(invalid))
+        invalid = self.get_texture_node_invalid(instance)
+        if invalid:
+            raise RuntimeError("At least a Albedo texture file"
+                               "nodes need to be connected")
+        invalid = self.get_texture_name_invalid(instance)
+        if invalid:
+            raise RuntimeError("Invalid texture name(s) found: "
+                               "{0}".format(invalid))
+
+    def get_texture_name_invalid(self, instance):
+
+        invalid = set()
+        shading_grp = self.get_material_from_shapes(instance)
+
+        # get the materials related to the selected assets
+        # get the file textures related to the PBS Shader
+        # validate the names of the textures
+
+        main_shader = cmds.listConnections(shading_grp,
+                                           destination=True,
+                                           type="StingrayPBS")
+        for shader in main_shader:
+            albedo = cmds.listConnections(shader + ".TEX_color_map")[0]
+            dif_path = cmds.getAttr(albedo + ".fileTextureName")
+            dif = dif_path.split(".")[0]
+            # "_D"
+            if not dif.endswith("_D"):
+                invalid.add(dif_path)
+            orm_packed = cmds.listConnections(shader + ".TEX_ao_map")
+            if orm_packed:
+                orm_path = cmds.getAttr(orm_packed[0] + ".fileTextureName")
+                orm = orm_path.split(".")[0]
+                # "_ORM"
+                if not orm.endswith("_ORM"):
+                    invalid.add(orm_path)
+            nrm = cmds.listConnections(shader + ".TEX_normal_map")
+            if nrm:
+                nrm_path = cmds.getAttr(nrm[0] + ".fileTextureName")
+                nrm_map = nrm_path.split(".")[0]
+                # "_N"
+                if not nrm_map.endswith("_N"):
+                    invalid.add(nrm_path)
+
+        return list(invalid)
+
+    def get_texture_node_invalid(self, instance):
+        invalid = set()
+        shading_grp = self.get_material_from_shapes(instance)
+        main_shader = cmds.listConnections(shading_grp,
+                                           destination=True,
+                                           type="StingrayPBS")
+        for shader in main_shader:
+            # diffuse texture file node
+            albedo = cmds.listConnections(shader + ".TEX_color_map")
+            if not albedo:
+                invalid.add(albedo)
+            orm_packed = cmds.listConnections(shader + ".TEX_ao_map")
+            if not orm_packed:
+                continue
```

Comment:
Continuing here means we would not run any of the `TEX_normal_map` logic below.

However, looking at the code it seems that since we're just using `continue` here the logic basically does nothing of note at all.

Reply:
I added this mainly because NoneType Error.